kura/test/org.eclipse.kura.container.orchestration.provider.test/src/test/java/org/eclipse/kura/container/orchestration/provider/ContainerOrchestrationServiceImplTest.java

```diff
@@ -265,30 +277,44 @@ public void testStopContainer() throws KuraException, InterruptedException {
 
         thenTestIfNewContainerDoesNotExists();
     }
-    
+
     @Test
     public void testDeleteImage() throws KuraException, InterruptedException {
         givenFullProperties(true);
         givenDockerServiceImplSpy();
         givenDockerClient();
-        
+
         whenImageIsDeletedById("y456y5146hrth");
-        
+
         thenCheckIfImageWasDeleted();
     }
-    
+
     @Test
     public void testListImage() throws KuraException, InterruptedException {
         givenFullProperties(true);
         givenDockerServiceImplSpy();
         givenDockerClient();
-        
+
         whenMockForImageListing();
-        
+
         whenImagesAreListed();
-        
+
         thenCheckIfImagesWereListed();
-        
+
+    }
+
+    @Test
+    public void testImageDigestsByContainerName() throws KuraException, InterruptedException {
+        givenFullProperties(true);
+        givenDockerServiceImplSpy();
+        givenDockerClient();
+
+        whenMockForImageDigestsListing();
+
+        whenGetImageDigestsByContainerName(IMAGE_NAME_NGINX);
+
+        thenDigestsListEqualsExpectedOne(EXPECTED_DIGESTS_ARRAY);
```

Comment:
Readability note: this is somewhat awkward to read since we already have the expectation as a parameter of the method. You could just have:

```suggestion
        thenDigestsListEquals(EXPECTED_DIGESTS_ARRAY);
```

which reads way better.